packages/shared-ui/src/flowHandler/processHandler.ts

```diff
@@ -91,6 +91,21 @@ export class ProcessHandler {
       return res;
     }
 
+    const block = res.val.blockBody.block;
+
+    // if the frontend preferred block is not in the initScreenBlocks, we need to init the signup process
+    // another condition to check that the new block is not in the initScreenBlocks is because in social login we come back to the signup process with addional data
+    if (
+      (!frontendPreferredBlockType || initScreenBlocks.includes(frontendPreferredBlockType)) &&
+      !initScreenBlocks.includes(block as BlockTypes)
+    ) {
+      res = await this.#corbadoApp.authProcessService.init(this.#abortController, BlockType.SignupInit, false, true);
```

Comment:
Why are we preferring signup in that case, shouldn't we set frontendPreferredBlockType to null in this case? (because the frontend does not have any preference for signup or login in this case)